src/backend/PublicInterfaceImpl.c

```diff
@@ -239,7 +237,9 @@ int BSL_API_ApplySecurity(const BSL_LibCtx_t *bsl, BSL_SecurityResponseSet_t *re
                 {
                     BSL_LOG_WARNING("Deleting bundle due to block target num %lu security failure",
                                     sec_oper->target_block_num);
-                    must_drop = true;
+                    // Drop the bundle and return operation error
+                    BSL_LOG_WARNING("***** Delete bundle due to failed security operation *******");
+                    BSL_BundleCtx_DeleteBundle(bundle);
```

Comment:
Maybe there is already an open ticket. This belongs as a PP finalize decision, not a BSL user-uncontrolled behavior. If it's not dealt with in this PR it needs a ticket.